crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -914,11 +914,11 @@ where
     T: Bundle + 'static,
 {
     fn write(self, world: &mut World) {
-        if let Some(mut entity) = world.get_entity_mut(self.entity) {
-            entity.insert(self.bundle);
-        } else {
-            panic!("error[B0003]: Could not insert a bundle (of type `{}`) for entity {:?} because it doesn't exist in this World.", std::any::type_name::<T>(), self.entity);
-        }
+        world.get_entity_mut(self.entity)
+            .unwrap_or_else(|| {
+                panic!("error[B0003]: Could not insert a bundle (of type `{}`) for entity {:?} because it doesn't exist in this World.", std::any::type_name::<T>(), self.entity);
+            })
```

Comment:
i would suggest `expect(format!())` style here but there may be a runtime cost of formatting every time regardless of whether we're going to panic or not

Reply:
The perf impact is precisely why it's using unwrap_or_else here. We shouldn't be allocating when we aren't panicking.